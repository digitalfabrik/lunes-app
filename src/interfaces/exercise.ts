export interface IVocabularyOverviewScreen {
  navigation: any
  route: any
}

export interface IVocabularyTrainerScreen {
  navigation: any
  route: any
}

export interface IAlternativeWordProps {
  alt_word: string
  article: string
}

export interface IDocumentProps {
  id: number
  word: string
  article: string
  document_image: Array<{ id: number; image: string }>
  audio: string
  alternatives: IAlternativeWordProps[]
}

export interface IVocabularyOverviewListItemProps {
  id: number
  word: string
  article: string
  image: string
  audio: string
}

export interface IConfirmationModalProps {
  visible: boolean
  setIsModalVisible: Function
  navigation: any
  extraParams: any
}

export interface IAnswerSectionProps {
  tryLater: () => void
  currentDocumentNumber: number
  setCurrentDocumentNumber: Function
  documents: IDocumentProps[]
  finishExercise: Function
  trainingSet: string
  disciplineTitle: string
}

export interface IFeedbackProps {
  secondAttempt: boolean
  result: string
  document?: IDocumentProps
  input: string
}

export interface IActionsProps {
  tryLater: () => void
  giveUp: () => void
  result: string
  checkEntry: () => void
  getNextWord: () => void
  input: string
  isFinished: boolean
  secondAttempt: boolean
}
