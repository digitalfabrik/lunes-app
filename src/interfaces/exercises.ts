export interface IExercisesScreenProps {
  route: any
  navigation: any
}
