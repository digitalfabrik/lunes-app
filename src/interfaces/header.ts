export interface IHeaderProps {
  top: number | undefined
}
