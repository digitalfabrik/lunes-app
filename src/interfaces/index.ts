import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
