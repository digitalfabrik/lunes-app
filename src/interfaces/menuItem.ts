import { ReactElement } from 'react'

export interface IMenuItemProps {
  selected: boolean
  onPress: () => void
  icon: string
  title: string
  children: ReactElement
}
