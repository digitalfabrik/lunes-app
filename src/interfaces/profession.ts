export interface IProfessionScreenProps {
  navigation: any
}

export interface IProfessionsProps {
  id: number
  title: string
  description: string
  icon?: string
  total_training_sets: number
}
