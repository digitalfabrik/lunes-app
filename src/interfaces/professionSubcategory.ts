export interface IProfessionSubcategoryScreenProps {
  route: any
  navigation: any
}

export interface IProfessionSubcategoryProps {
  id: number
  title: string
  description: string
  icon: string
  total_documents: number
}
