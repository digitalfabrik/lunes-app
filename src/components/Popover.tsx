import React from 'react'
import Popover, { PopoverPlacement } from 'react-native-popover-view'
import { IPopoverProps } from '../interfaces'
import { StyleSheet } from 'react-native'
import { COLORS } from '../constants/colors'

export const styles = StyleSheet.create({
  arrow: {
    backgroundColor: COLORS.lunesBlack
  },
  overlay: {
    backgroundColor: 'transparent'
  }
})
const AlertPopover = React.forwardRef(({ children, isVisible, setIsPopoverVisible }: IPopoverProps, ref) => (
  <Popover
    isVisible={isVisible}
    onRequestClose={() => setIsPopoverVisible(false)}
    from={ref}
    placement={PopoverPlacement.TOP}
    arrowStyle={styles.arrow}
    arrowShift={-0.85}
    verticalOffset={-10}
    backgroundStyle={styles.overlay}>
    {children}
  </Popover>
))

export default AlertPopover
