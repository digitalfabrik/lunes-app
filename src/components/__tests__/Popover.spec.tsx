import { Text } from 'react-native'
import React from 'react'
import Popover from '../Popover'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { IPopoverProps } from '../../interfaces'

describe('Components', () => {
  describe('Popover', () => {
    const defaultPopoverProps: IPopoverProps = {
      children: <Text>children</Text>,
      isVisible: true,
      setIsPopoverVisible: () => {}
    }

    it('renders correctly across screens', () => {
      const component = shallow(<Popover {...defaultPopoverProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should have visible property passed to it as default', () => {
      const popoverProps: IPopoverProps = {
        ...defaultPopoverProps
      }

      const component = shallow(<Popover {...popoverProps} />)
      expect(component.props().isVisible).toBe(true)
    })

    it('should have visible property passed to it', () => {
      const popoverProps: IPopoverProps = {
        ...defaultPopoverProps,
        isVisible: false
      }

      const component = shallow(<Popover {...popoverProps} />)
      expect(component.props().isVisible).toBe(false)
    })

    it('should render children passed to it', () => {
      const component = shallow(<Popover {...defaultPopoverProps} />)
      expect(component.contains(defaultPopoverProps.children)).toBe(true)
    })
  })
})
