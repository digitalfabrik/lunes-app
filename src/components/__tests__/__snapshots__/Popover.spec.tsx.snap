// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Popover renders correctly across screens 1`] = `
<Popover
  arrowShift={-0.85}
  arrowStyle={
    Object {
      "backgroundColor": "#000e38",
    }
  }
  backgroundStyle={
    Object {
      "backgroundColor": "transparent",
    }
  }
  debug={false}
  from={null}
  isVisible={true}
  mode="rn-modal"
  onCloseComplete={[Function]}
  onCloseStart={[Function]}
  onOpenComplete={[Function]}
  onOpenStart={[Function]}
  onRequestClose={[Function]}
  placement="top"
  verticalOffset={-10}
>
  <Text>
    children
  </Text>
</Popover>
`;
